### PR TITLE
Attempts to bias correct and pulling out functions from `03` script

### PR DESCRIPTION
**Pull request by SamStruthers**
Hey B and Juan!
I gave a pass at the bias correction options (`src/apply_ml_bias_correction`) discussed in the paper from MR, as we discussed in the working meeting/teams, it does not appear to be that useful. This is mainly due to the lack of samples we have the work with and inconsistently in model predictions. @juandlt-csu I have not evaluated your point about aggregating all the val sets to train the ML2-Res model, mostly because I want to get this code out and i think it would take a bit of reworking. 
This PR mostly has to do with optimizing the 03 script in a couple ways:
- I pulled out the function `apply_training_scale` into it's own function so that it can be used for other modelling. This is part of #40 
- I pulled out the function to plot timeseries model input/output timeseries into `plot_model_timeseries`. This will effectively get rid of the `04` script (see end of 03 script). and it is part of #40 
- I was running into some reproducibility issues with `xgboost_hyperparameter_tuning` but some small adjustments to the method and setting seed within the function seem to have corrected that moving forward. I also pulled out the user console interaction and had to add some code to the 03 script (ln:736-790) to select the exact models. 

Other things to note:
-`02`: couple of small tweaks based on the new data structure from `load_grab_sample_data`
- `generate_fc_grab_sample_data`: This was not correctly removing spaces from the site_code names so i went ahead and fixed that. 
- `load_grab_sample_data`: updating file paths and adding sourcing of `read_ext`
- `setup_ross_theme`: I was running into some issues with my plotting functions since I was using ROSS_theme so I made a quick script to source for other other plotting functions. I then removed this from the setup_libraries script

The 03 script does knit so if you'd prefer to see a knitted version, let me know!

Review Req:
Juan:
Can you check out the new `src` scripts to make sure I am not missing anything glaring or there are some inconsistencies? They appear to work as expected but want to make sure they will be reproducible for other models (ie realtime chl-a)
B: Can you check out the general flow of the `03` script to make sure I am not missing any important steps? 

Next Steps: 
- I think the `load_grab_sample` needs to be optimized/filepaths hardcoded, it was quick difficult to use. I know Juan is shifting his focus to NPS, so I can take care of that next week and should be a easy fix.

### CODE REVIEW

File: modeling/toc/03_ross_only_TOC_model.Rmd [hunk 1]
```diff
@@ -32,9 +36,15 @@ source("src/plot_tv_fold_perf.R")
...
+#load grab sample data function
+source("src/load_grab_sample_data.R")
+#function to plot model timeseries
+source("src/plot_model_timeseries.R")
 ```
 
-# Create scaling based on all sensor data
+## Read in sensor dataset
 
 We should functionalize this chunk so that we can easily apply the scaling as we get new sensor data in the future.
```
Comment (steeleb) [inline modeling/toc/03_ross_only_TOC_model.Rmd]:
```suggestion
Here, we read in the sensor data, apply min-max scaling defined in the `apply_sensor_transformations_toc()` function, and store the dataset for use later
```

File: modeling/toc/03_ross_only_TOC_model.Rmd [hunk 2]
```diff
@@ -841,6 +879,114 @@ final_plot
...
+
+rm(list = setdiff(ls(), c("train_val", "testing", "full_model","version","id_cols",
+                          "complete_sensor_data",
+                          #keeping functions around
+                          "apply_training_scale","plot_tvt_fold_perf","plot_tvt_ensemble_perf",
+                          "apply_ml_bias_correction", "load_grab_sample_data", "plot_model_timeseries" )) )
+```
+
+
+# Bias Correction
```
Comment (steeleb) [inline modeling/toc/03_ross_only_TOC_model.Rmd]: I'd suggest popping out the bias correction into it's own script.

File: src/plot_model_timeseries.R [hunk 3]
```diff
@@ -0,0 +1,363 @@
...
+  #process model input data
+  # Summarize data to desired interval and trim to individual site
+  summarized_data <- model_input_df%>%
+    filter(site == site_sel)%>%
+    mutate(DT_round = round_date(DT_round, unit = summary_interval )) %>% # ensure DT_round is rounded to the nearest
+    group_by(site, DT_round) %>%
+    summarise(across(everything(), mean, na.rm = TRUE), .groups = 'drop')
+
+  if(method  %in% c("Ensemble", "ensemble")){
+    #browser()
```
Comment (steeleb) [inline src/plot_model_timeseries.R]:
```suggestion
```

File: modeling/toc/03_ross_only_TOC_model.Rmd (hunk 1 above)
Comment (SamStruthers) [reply modeling/toc/03_ross_only_TOC_model.Rmd]: Thank you!!
